main_test.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestWordOccurrences(t *testing.T) {
+	for _, tc := range []struct {
+		name       string
+		input      string
+		wantstring []string
+		wantuint   []uint
+	}{
+		{"empty", "", nil, nil},
```

Comment:
Other useful tests:
 * all unique words
 * multiple occurrences of the same word
 * all the same words
 * a non-space separator

Reply:
The tests are good now.